src/Microsoft.AspNet.Mvc.Core/RemoteAttribute.cs

```diff
@@ -0,0 +1,238 @@
+        /// </summary>
+        /// <param name="property">One field name the client should include in a validation request.</param>
+        /// <returns>Name of a field the client should include in a validation request.</returns>
+        /// <remarks>Returns <paramref name="property"/> with a <c>"*."</c> prefix.</remarks>
+        public static string FormatPropertyForClientValidation(string property)
+        {
+            if (string.IsNullOrEmpty(property))
+            {
+                throw new ArgumentException(Resources.ArgumentCannotBeNullOrEmpty, "property");
+            }
+
+            return "*." + property;
+        }
+
+        /// <summary>
+        /// Returns the URL where the client should send a validation request.
+        /// </summary>
+        /// <param name="context">The <see cref="ClientModelValidationContext"/> used to generate the URL.</param>
+        /// <returns>The URL where the client should send a validation request.</returns>
+        protected virtual string GetUrl([NotNull] ClientModelValidationContext context)
+        {
+            var urlHelper = context.RequestServices.GetRequiredService<IUrlHelper>();
+            var url = urlHelper.RouteUrl(RouteName, values: RouteData, protocol: null, host: null, fragment: null);
+            if (url == null)
+            {
+                throw new InvalidOperationException(Resources.RemoteAttribute_NoUrlFound);
+            }
+
+            return url;
+        }
+
+        /// <inheritdoc />
+        public override string FormatErrorMessage(string name)
+        {
+            return string.Format(CultureInfo.CurrentCulture, ErrorMessageString, name);
+        }
+
+        /// <inheritdoc />
+        public override bool IsValid(object value)
+        {
+            return true;
+        }
+
+        /// <inheritdoc />
+        /// <exception cref="InvalidOperationException">
+        /// Thrown if unable to generate a target URL for a validation request.
+        /// </exception>
+        public virtual IEnumerable<ModelClientValidationRule> GetClientValidationRules(
+            [NotNull] ClientModelValidationContext context)
+        {
+            var metadata = context.ModelMetadata;
+            var rule = new ModelClientValidationRemoteRule(
+                FormatErrorMessage(metadata.GetDisplayName()),
+                GetUrl(context),
+                HttpMethod,
+                FormatAdditionalFieldsForClientValidation(metadata.PropertyName));
+
+            return new[] { rule };
+        }
+
+        private static IEnumerable<string> SplitString(string original)
+        {
+            if (string.IsNullOrEmpty(original))
+            {
+                return new string[0];
+            }
+
+            var split = original.Split(',')
+                                .Select(piece => piece.Trim())
+                                .Where(trimmed => !string.IsNullOrEmpty(trimmed));
+            return split;
+        }
```

Comment:
because property names containing any whitespace at all causes problems everywhere else.  want the trimmed entries as well as to eliminate null or whitespace ones.